### PR TITLE
Improve subtitle behaviour using PlayerMonitor

### PR DESCRIPTION
**Pull request by mediaminister**
This includes:
- Implement PlayerMonitor service
- Check subtitles `onAVStarted` and `onAVChange`:
  - Select internal InputStream Adaptive subtitles if available
  - Add external recalculated subtitle if no subtitles available
  - Enable subtitles if `showsubtitles` add-on setting is enabled

There are still bugs in InputStream Adaptive or Kodi that cause the following problems:
- ~Internal subtitles suddenly disappear, seeking fixes this~ fixed in https://github.com/peak3d/inputstream.adaptive/pull/456
- External subtitles get out of sync when a new period starts

### CODE REVIEW

File: resources/lib/vtmgo/vtmgostream.py [hunk 1]
```diff
@@ -270,6 +270,7 @@ def _delay_subtitles(self, subtitles, json_manifest):
 
         for subtitle in subtitles:
             output_file = temp_dir + '/' + subtitle.split('/')[-1].split('.')[0] + '.nl-NL.' + subtitle.split('.')[-1]
+            output_file = temp_dir + '/' + 'VTM GO.' + subtitle.split('.')[-1]
```
Comment (michaelarnauts) [inline resources/lib/vtmgo/vtmgostream.py:273]: I think one of these lines should be removed.

**Comment by dagwieers** [comment]
Personally I don't like it that add-ons need a service and a player monitor (running all the time) just to ensure that e.g. the subtitles work correctly. So it is unfortunate that we have to work around it like this.

That said, if this is a temporary measure (and we ensure things get fixed properly upstream at some point in the future) at least things work better for end-users, which is paramount.

**Comment by mediaminister** [comment]
> Personally I don't like it that add-ons need a service and a player monitor (running all the time) just to ensure that e.g. the subtitles work correctly. So it is unfortunate that we have to work around it like this.

There are too many problems with subtitles which we cannot quickly fix ourselves, so manually switching between external en internal subtitles will be needed for a long time I think:
- Sometimes VTM GO MPD manifests only define internal subtitles for the first period (e.g. 13 Geboden episode 2). After inspecting the actual stream, it does contain subtitles for the whole program but this is not detected in InputStream Adaptive.
- InputStream Adaptive suddenly stops displaying internal WEBVTT subtitles in some cases, fixed in https://github.com/peak3d/inputstream.adaptive/pull/456
- External subtitles get out of sync when a new period starts, this is a rather difficult bug to fix as you need to understand Kodi Video Player source
- Kodi's `setSubtitles` API automatically enables subtitles and sometimes this is unwanted behaviour because we only want to add an external subtitle file without overruling a users choice to disable subtitles.

**Comment by dagwieers** [comment]
@mediaminister My comment wasn't criticizing this PR specifically, but rather a general notice. However I prefer if we open a tracking issue for each of the issues and report them upstream now things are still fresh.

*Interesting, VTM GO uses the Anvato Universal Player, would be nice to add it to our testing wiki page...*

**Comment by dagwieers** [comment]
We started watching Matroesjkas, which without subtitles is quite impossible with Russian or Lithuanian conversations.
Test-driving these changes, so far so good.

**Comment by dagwieers** [comment]
Using PR #183 I get the following error trying to play movie *Gulliver's Travel*.

```log
2020-07-03 19:36:11.345 T:1042281344  NOTICE: [plugin.video.vtm.go] [vtmgostream] Downloading text from https://dcs-vod.apis.anvato.net/vod/p/session/manifest.mpd?i=i177999829-ndbc68579-a470-4a83-995d-8bdb27af72e1
2020-07-03 19:36:11.431 T:1042281344  NOTICE: [plugin.video.vtm.go] [urllib3.connectionpool] https://dcs-vod.apis.anvato.net:443 "GET /vod/p/session/manifest.mpd?i=i177999829-ndbc68579-a470-4a83-995d-8bdb27af72e1 HTTP/1.1" 200 None
2020-07-03 19:36:11.552 T:1042281344   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.AttributeError'>
                                            Error Contents: 'NoneType' object has no attribute 'get'
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.vtm.go/plugin.py", line 16, in <module>
                                                plugin.run(sys.argv)
                                              File "/storage/.kodi/addons/plugin.video.vtm.go/resources/lib/plugin.py", line 218, in run
                                                routing.run(params)
                                              File "/storage/.kodi/addons/script.module.routing/lib/routing.py", line 130, in run
                                                self._dispatch(self.path)
                                              File "/storage/.kodi/addons/script.module.routing/lib/routing.py", line 141, in _dispatch
                                                view_func(**kwargs)
                                              File "/storage/.kodi/addons/plugin.video.vtm.go/resources/lib/plugin.py", line 199, in play
                                                Player(kodi).play(category, item)
                                              File "/storage/.kodi/addons/plugin.video.vtm.go/resources/lib/modules/player.py", line 59, in play
                                                resolved_stream = self._vtm_go_stream.get_stream(category, item)
                                              File "/storage/.kodi/addons/plugin.video.vtm.go/resources/lib/vtmgo/vtmgostream.py", line 102, in get_stream
                                                subtitles = self._extract_subtitles_from_stream_info(stream_info)
                                              File "/storage/.kodi/addons/plugin.video.vtm.go/resources/lib/vtmgo/vtmgostream.py", line 213, in _extract_subtitles_from_stream_info
                                                name = '{} - {}_{}'.format(stream_info.get('video').get('metadata').get('program').get('title'), stream_info.get('video').get('metadata').get('title'), idx)
                                            AttributeError: 'NoneType' object has no attribute 'get'
                                            -->End of Python script error report<--
```

**Comment by mediaminister** [comment]
Fixed again.

**Comment by michaelarnauts** [comment]
I think this branch is better then what we have right now. Thanks @mediaminister!

**Pull request merged**